trl/trainer/grpo_config.py

```diff
@@ -112,9 +112,19 @@ class GRPOConfig(TrainingArguments):
         scale_rewards (`bool`, *optional*, defaults to `True`):
             Whether to scale the rewards by dividing them by their standard deviation. If `True` (default), the rewards
             are normalized by the standard deviation, ensuring they have unit variance. If `False`, no scaling is
-            applied. The [Dr. GRPO](https://github.com/sail-sg/understand-r1-zero/blob/main/understand-r1-zero.pdf)
-            paper recommends not scaling the rewards, as scaling by the standard deviation introduces a question-level
-            difficulty bias.
+            applied. The [Dr. GRPO paper](https://huggingface.co/papers/2503.14476) recommends not scaling the rewards,
+            as scaling by the standard deviation introduces a question-level difficulty bias.
+        loss_type (`str`, *optional*, defaults to `"bnpo"`):
+            Specifies the loss formulation to use. Supported values are:
+
+            - `"bnpo"`: Token-level losses are aggregated by normalizing with the completion length
+                within the local batch. Note that normalization is performed over the local batch only, so results may
+                slightly vary depending on the local batch size, despite a constant effective batch size.
+            - `"drgrpo"`: Token-level losses are aggregated by normalizing with a global constant. This method was
+                introduced in the [Dr. GRPO paper](https://huggingface.co/papers/2503.14476) to eliminate length bias.
+                The value of the constant corresponds to `max_completion_length`.
```

Comment:
If I understand correctly, @edbeeching was trying something slightly different in https://github.com/huggingface/trl/pull/3231 that did local scaling per batch instead of a global constant. Do you know if there's much difference between the two?

Reply:
They are roughly equivalent, I have closed my PR in favor of this one.